### DISCUSSION TITLE
`createUser` method returning `User not Allowed`

### DISCUSSION THREAD

**Discussion by Perfecto-always** [supabase/supabase#5434] [Questions] [answered]
I am creating user from an express backend like this:
![image](https://user-images.githubusercontent.com/74700079/153709918-07fdd311-267b-41b2-a12b-28f855f686d3.png)

but this is throwing an error of
![image](https://user-images.githubusercontent.com/74700079/153709979-667a4ba1-343d-42cc-a2a9-60fdab5ff6f9.png)

OS: `Windows`
Library Version: `^1.29.4`

Here are my authentication settings 
![image](https://user-images.githubusercontent.com/74700079/153710073-42cc9ec9-2e34-4480-ba43-8dfcbb10e678.png)

**Comment by GaryAustin1** [comment, marked as answer]
Are you using service_role key?   Error is you don’t have a key(user) with permission to access that function.

**Reply by Perfecto-always** [reply]
I got it working like this
```js
  const supbaseAdmin = createClient(
    process.env.SUPABASE_URL,
    process.env.SUPABASE_SERVICE_ROLE_KEY
  );

  const { data: user, error } = await supbaseAdmin.auth.api.createUser({
    email: "example@gmail.com",
    password: "password",
    data: { name: "Perfecto" },
  });
```

Thanks for the help 🙏🙂

**Reply by kVarunkk** [reply]
Hey, I'm using this same method to create an admin user but it still creates a user with the "authenticated" role. I can't figure out what's wrong!
![image](https://github.com/supabase/supabase/assets/98093422/a01efd50-9906-497f-b62d-b708c8e869f3)
![image](https://github.com/supabase/supabase/assets/98093422/de3333c1-5990-4217-9b0d-b89ab01e86f6)

**Comment by Yash-pede** [comment]
i am using next js and while importing 
![Screenshot from 2023-11-29 22-46-46](https://github.com/supabase/supabase/assets/116386431/4120112e-6539-46b7-955a-a24906916c90)


and If using this 
![image](https://github.com/supabase/supabase/assets/116386431/c042713a-9958-446a-a879-d59a443f0bdb)

it says 
`error AuthApiError: User not allowed
    at handleError (webpack-internal:///(app-pages-browser)/./node_modules/@supabase/gotrue-js/dist/module/lib/fetch.js:42:11)
    at async _handleRequest (webpack-internal:///(app-pages-browser)/./node_modules/@supabase/gotrue-js/dist/module/lib/fetch.js:79:9)
    at async _request (webpack-internal:///(app-pages-browser)/./node_modules/@supabase/gotrue-js/dist/module/lib/fetch.js:64:18)
    at async GoTrueAdminApi.listUsers (webpack-internal:///(app-pages-browser)/./node_modules/@supabase/gotrue-js/dist/module/GoTrueAdminApi.js:140:30)
    at async getUser (webpack-internal:///(app-pages-browser)/./app/(dashboard)/dashboard/users/page.tsx:20:48)`
    
    
    
![image](https://github.com/supabase/supabase/assets/116386431/0d7c36a8-1595-465c-abd5-0ad65d26fd36)



please help

**Reply by GaryAustin1** [reply]
You do not use auth-helpers/ssr clients for service role.  They will add the user session to the client which will override the key.
Just use createClient from supabase-js for that.
